### ISSUE TITLE
Problem with nested ui-views

### ISSUE THREAD

**Issue by Molnfront** [Gillardo/bootstrap-ui-treeview#3]
### Problem

I have tried to implement it in my multi view app that uses ui-router. I have put it in a nested ui-view:

```
<div ui-view="treeview" class="col-sm-4"></div>
```

No error and no tree view. I am using Angular 1.4.7

It could be the famous UI-Router problem where the directive is loaded before the ui-view is loaded. I tried to work around that whit a scope.$watch("$viewContentLoaded") but no luck. (http://stackoverflow.com/questions/24300387/check-if-a-sub-view-is-reloaded-angularjs-ui-router or http://stackoverflow.com/questions/24198786/custom-directives-not-active-in-nested-views-when-using-ui-router)

I have also added:

```
$rootScope.$on("$stateChangeError", console.log.bind(console));
```

So I can see any error.
### What I am trying to do

I have an admin UI with two views: a tree view to the left and a editor to the right. I tried this in my multi-view app so that I can programmatically refresh only the tree after the delete button is pressed in the right view and a file is removed.

If I put the tree in the main view it works but then I have no way to refresh the view. Not that I know of at least. 

I tried to use:

```
$state.go('app.dashboard');
```

which should update the main view but nothing happens.

I´m working on a plunker...

**Comment by Gillardo** [comment]
Let me know when u have a plunkr and i will take a look

**Comment by Molnfront** [comment]
Here is my plunkr that shows that the tree-view directive is not showing up in a nested ui-router config:

http://plnkr.co/edit/DZFTym?p=preview

If you inspect the view in your browser for the menu tree view you can see that the div tree-view is added, but no content inside the tree and no error!

In app.js I put the tree-view in the .state('tabs.first.firstTabSub1'

**Comment by Molnfront** [comment]
I got one of my experiment working: http://plnkr.co/edit/L5H4KZ
This time I tried with angularBootstrapNavTree, one of the updated forks. One problem I found is that in nested states the data that the directive use must be in the same state for the directive to find, so I first tried to put it in a resolve but only got errors. Then I tried to put it in a controller in the state in app.js and that worked. 

To also make it work with my REST service and not only a json data string, I had to both use a controller and a resolver in the state provider.

I haven´t got time to test it with your directive yet,
